test/filter_alignments_test.py

```diff
@@ -0,0 +1,219 @@
+import unittest
+
+import numpy as np
+import pandas as pd
```

Comment:
Why `filter_alignments_test`? Is this supposed to be doing the filtering of molecule tables? If so, let's name it differently from alignments. Alignments makes it sound like we're still working with strings, whereas we're actually working with sets of reads for intBCs & alleles.